WooCommerce/Classes/POS/Presentation/PointOfSaleEntryPointView.swift

```diff
@@ -15,26 +19,36 @@ struct PointOfSaleEntryPointView: View {
          collectOrderPaymentAnalyticsTracker: POSCollectOrderPaymentAnalyticsTracking) {
         self.onPointOfSaleModeActiveStateChange = onPointOfSaleModeActiveStateChange
 
-        let posModel = PointOfSaleAggregateModel(
-            itemsController: itemsController,
-            cardPresentPaymentService: cardPresentPaymentService,
-            orderController: orderController,
-            collectOrderPaymentAnalyticsTracker: collectOrderPaymentAnalyticsTracker)
-
-        self._posModel = State(wrappedValue: posModel)
+        self.itemsController = itemsController
+        self.cardPresentPaymentService = cardPresentPaymentService
+        self.orderController = orderController
+        self.collectOrderPaymentAnalyticsTracker = collectOrderPaymentAnalyticsTracker
     }
 
     var body: some View {
-        PointOfSaleDashboardView()
+        Group {
```

Comment:
Cleverly done! When I experimented I didn't consider hiding it behind `PointOfSaleLoadingView()` until `posModel` was initialized and only then passing environment.